HelpMyStreetFE/HelpMyStreetFE/Controllers/AccountController.cs

```diff
@@ -7,27 +7,79 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
 using Microsoft.AspNetCore.Authorization;
+using System.Security.Claims;
+using HelpMyStreetFE.Services;
 
 namespace HelpMyStreetFE.Controllers
 {
     [Authorize]
     public class AccountController : Controller
     {
         private readonly ILogger<AccountController> _logger;
-        public AccountController(ILogger<AccountController> logger)
+        private readonly IUserService _userService;
+        public AccountController(
+            ILogger<AccountController> logger,
+            IUserService userService
+            )
         {
             _logger = logger;
+            _userService = userService;
+        }
+
+        private UserDetails GetUserDetails(HelpMyStreet.Utils.Models.User user)
+        {
+            var personalDetails = user.UserPersonalDetails;
+            string initials = personalDetails.FirstName.Substring(0, 1) + personalDetails.LastName.Substring(0, 1);
+            string address = personalDetails.Address.AddressLine1 + "," + personalDetails.Address.Postcode;
+            string gender = "XXX";
+            bool underlyingMedicalConditions = false;
+
+            if(personalDetails.UnderlyingMedicalCondition.HasValue)
+            {
+                underlyingMedicalConditions = personalDetails.UnderlyingMedicalCondition.Value;
+            }
+                    
+            return new UserDetails(
+                initials,
+                personalDetails.DisplayName,
+                personalDetails.FirstName,
+                personalDetails.LastName,
+                personalDetails.EmailAddress,
+                address,
+                personalDetails.MobilePhone,
+                personalDetails.OtherPhone,
+                personalDetails.DateOfBirth.Value.ToString(),
+                gender,
+                underlyingMedicalConditions
+                );
         }
 
         [HttpGet]
         public IActionResult Index()
         {
             //TODO: Check that the user is authenticated
+            var id = int.Parse(HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value);
+            HelpMyStreet.Utils.Models.User user = _userService.GetUserAsync(id).Result;
```

Comment:
Change the signature to `public async Task<IActionResult> Index()` and then await the call so this doesn't lock the thread

Reply:
Also add a using for `HelpMyStreet.Utils.Models` so you don't need to declare the full namespace here?